pkg/cli/project/project.go

```diff
@@ -320,18 +307,43 @@ func (o *ProjectOptions) GetContextFromName(contextName string) (*clientcmdapi.C
 }
 
 func ConfirmProjectAccess(currentProject string, projectClient projectv1client.ProjectV1Interface, kClient corev1client.CoreV1Interface) error {
-	_, projectErr := projectClient.Projects().Get(context.TODO(), currentProject, metav1.GetOptions{})
-	if !kapierrors.IsNotFound(projectErr) && !kapierrors.IsForbidden(projectErr) {
-		return projectErr
-	}
+	projects, err := GetProjects(projectClient, kClient)
+	if err != nil {
+		_, projectErr := projectClient.Projects().Get(context.TODO(), currentProject, metav1.GetOptions{})
+		if !kapierrors.IsNotFound(projectErr) && !kapierrors.IsForbidden(projectErr) {
+			return projectErr
+		}
 
-	// at this point we know the error is a not found or forbidden, but we'll test namespaces just in case we're running on kube
-	if _, err := kClient.Namespaces().Get(context.TODO(), currentProject, metav1.GetOptions{}); err == nil {
+		// at this point we know the error is a not found or forbidden, but we'll test namespaces just in case we're running on kube
+		if _, err = kClient.Namespaces().Get(context.TODO(), currentProject, metav1.GetOptions{}); err == nil {
+			return nil
+		}
+	}
+	matched := false
+	for _, p := range projects {
+		if p.Name == currentProject {
+			matched = true
+			break
+		}
+	}
+	if matched {
 		return nil
 	}
-
+	var msg string
+	switch len(projects) {
```

Comment:
Original code iterates through projects only when `GetProjects`'s err is nil. This changes adds another case where err  is either IsNotFound or IsForbidden but the namespace does not exist. In which case `projects` is nil. Which is still aligned with the comment and the error message for case 0.